edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/EdgeOperator.cs

```diff
@@ -861,6 +894,47 @@ VolumeOptions GetVolumesFromModule(IModule<AgentDocker.CombinedDockerConfig> mod
                 : Option.None<(List<V1Volume>, List<V1VolumeMount>, List<V1VolumeMount>)>();
         }
 
+        private async Task GetOrCreatePersistentVolumeClaim(string name, bool readOnly)
+        {
+            var listResult = await this.client.ListNamespacedPersistentVolumeClaimAsync(KubeUtils.K8sNamespace);
+
+            var foundPvc = listResult.Items.SingleOrDefault(item => !string.IsNullOrWhiteSpace(item.Metadata.Name) && item.Metadata.Name == name);
+
+            if (foundPvc != default(V1PersistentVolumeClaim))
+            {
+                // ReadWriteOnce is the Kubernetes moniker for single pod use only
+                if (foundPvc.Spec.AccessModes.Contains("Once"))
+                {
+                    // TODO : should we throw here or should we just let Kube throw when we try to mount it?
+                    throw new AccessViolationException($"PVC with name {name} is configured to only be mounted to one node.");
```

Comment:
Why do we care about the access mode?

Reply:
Changed this to a warning - if the user sets this up wrong, we will probably see either exceptions or the pod will just stay pending..